tests/expressions_window/tests.py

```diff
@@ -0,0 +1,658 @@
+            order_by='salary',
+            partition_by='department')
+        )
+
+        self.assertQuerysetEqual(qs, [
+            ('Williams', 'Accounting', 37000, 37000),
+            ('Jenson', 'Accounting', 45000, 37000),
+            ('Jones', 'Accounting', 45000, 37000),
+            ('Adams', 'Accounting', 50000, 37000),
+            ('Moore', 'IT', 34000, 34000),
+            ('Wilkinson', 'IT', 60000, 34000),
+            ('Johnson', 'Management', 80000, 80000),
+            ('Miller', 'Management', 100000, 80000),
+            ('Smith', 'Marketing', 38000, 38000),
+            ('Johnson', 'Marketing', 40000, 38000),
+            ('Brown', 'Sales', 53000, 53000),
+            ('Smith', 'Sales', 55000, 53000),
+        ], lambda row: (row.name, row.department, row.salary, row.min_salary), ordered=False)
+
+    @skipUnlessSupportsFunction('Max')
+    def test_max_per_year(self):
+        qs = WindowTestModel.objects.annotate(max_salary_year=Window(
+            expression=Max('salary'),
+            partition_by=ExtractYear('hiredate'), )
+        )
+
+        self.assertQuerysetEqual(qs, [
+            ('Jones', 'Accounting', 45000, 2005, 100000),
+            ('Johnson', 'Management', 80000, 2005, 100000),
+            ('Miller', 'Management', 100000, 2005, 100000),
+            ('Smith', 'Sales', 55000, 2007, 55000),
+            ('Jenson', 'Accounting', 45000, 2008, 45000),
+            ('Williams', 'Accounting', 37000, 2009, 53000),
+            ('Smith', 'Marketing', 38000, 2009, 53000),
+            ('Brown', 'Sales', 53000, 2009, 53000),
+            ('Wilkinson', 'IT', 60000, 2011, 60000),
+            ('Johnson', 'Marketing', 40000, 2012, 40000),
+            ('Adams', 'Accounting', 50000, 2013, 50000),
+            ('Moore', 'IT', 34000, 2013, 50000),
+        ], lambda row: (row.name, row.department, row.salary, row.hiredate.year, row.max_salary_year,), ordered=False)
+
+    @skipUnlessSupportsFunction('CumeDist')
+    def test_cume_dist(self):
+        qs = WindowTestModel.objects.annotate(cume_dist=Window(
+            expression=CumeDist(),
+            order_by='salary')
+        )
+
+        self.assertQuerysetEqual(qs, [
+            ('Moore', 'IT', 34000, 0.0833333333333333),
+            ('Williams', 'Accounting', 37000, 0.166666666666667),
+            ('Smith', 'Marketing', 38000, 0.25),
+            ('Johnson', 'Marketing', 40000, 0.333333333333333),
+            ('Jones', 'Accounting', 45000, 0.5),
+            ('Jenson', 'Accounting', 45000, 0.5),
+            ('Adams', 'Accounting', 50000, 0.583333333333333),
+            ('Brown', 'Sales', 53000, 0.666666666666667),
+            ('Smith', 'Sales', 55000, 0.75),
+            ('Wilkinson', 'IT', 60000, 0.833333333333333),
+            ('Johnson', 'Management', 80000, 0.916666666666667),
+            ('Miller', 'Management', 100000, 1),
+        ], lambda row: (row.name, row.department, row.salary, row.cume_dist), ordered=False)
```

Comment:
Oracle round `CumeDist` with greater then other db precision. I think that we can add `round` for test unification. 
```diff
diff --git a/tests/expressions_window/tests.py b/tests/expressions_window/tests.py
index 924928c..7982823 100644
--- a/tests/expressions_window/tests.py
+++ b/tests/expressions_window/tests.py
@@ -318,9 +318,8 @@ class TestWindowFunction(TestCase):
             expression=CumeDist(),
             order_by='salary')
         )
-
         self.assertQuerysetEqual(qs, [
-            ('Moore', 'IT', 34000, 0.0833333333333333),
+            ('Moore', 'IT', 34000, 0.083333333333333),
             ('Williams', 'Accounting', 37000, 0.166666666666667),
             ('Smith', 'Marketing', 38000, 0.25),
             ('Johnson', 'Marketing', 40000, 0.333333333333333),
@@ -332,7 +331,7 @@ class TestWindowFunction(TestCase):
             ('Wilkinson', 'IT', 60000, 0.833333333333333),
             ('Johnson', 'Management', 80000, 0.916666666666667),
             ('Miller', 'Management', 100000, 1),
-        ], lambda row: (row.name, row.department, row.salary, row.cume_dist), ordered=False)
+        ], lambda row: (row.name, row.department, row.salary, round(row.cume_dist, 15)), ordered=False)
 
     @skipUnlessSupportsFunction('NthValue')
     def test_nthvalue(self):
```